focal/4_gs1_identification.html

```diff
@@ -0,0 +1,229 @@
+<section>
+      <h3>Chain of GS1 Credentials to Identify a Trade Item</h3>
+      <h4>Background</h4>
+      <p>
+            This use case has been provided by <a href="https://www.gs1.org/" rel="nofollow">GS1</a>.
+      </p>
+      <p>
+            GS1 is the global supply chain standards development organization behind the retail barcode. The content of
+            the barcode, the Global Trade Item Number (GTIN), is a 13-digit string composed of a GS1 Company Prefix (a
+            unique string
+            of 4-12 digits), a trade item reference (a numeric string unique within the GS1 Company Prefix to bring the
+            length
+            up to 12 digits), and a check digit (a mathematical calculation to detect keying errors).
```

Comment:
Do I read this correctly? If the `GS1 Company Prefix` is the max length of 12 digits, then the trade item reference must be 0 digits "to bring the length up to 12 digits", so that company cannot have any products? 

Likewise, if the `GS1 Company Prefix` is the almost-max length of 11 digits, then the trade item reference must be 1 digit "to bring the length up to 12 digits", and thus the company can have only 10 products (digits 0-9)?

Something seems rather wrong here.

Reply:
It's an edge case for the GS1 Company Prefix and reflects some legacy business practices. It still works, though.

If you have a 7-digit GS1 Company Prefix, you have 5 digits remaining for the trade item reference and 1 for the check digit. The total number of possible GTINs is 100,000, where the trade item reference ranges from "00000" to "99999".

Mathematically, if the length of the GS1 Company Prefix is N, then the total number of possible GTINs is 10^(12-N). In the edge case where N=12, the total number of possible GTINs is 10^0=1, where the trade item reference is an empty string. Despite the fact that we refer to many of our keys as numbers (GTIN = Global Trade Item Number), they are in fact strings composed entirely of digits. This avoids the ambiguity where the string value starts with one or more zeros. Because of this, an empty string is a legitimate trade item / location / asset / document / coupon reference for a 12-digit GS1 Company Prefix.

And yes, you can have one of each type of key, all with the same value, but all distinct. The type of key is determined either by the context (e.g., in an EAN/UPC barcode it can only be a GTIN, in field X in an EDI document it can only be a GLN) or stated explicitly (e.g., using numeric codes called Application Identifiers in larger barcodes or in URIs, encoding bits in RFID tags).

I'm not personally aware of any 12-digit GS1 Company Prefixes being issued today, but they have been in the past. As for 11 digits, that can be done for companies that have only a very few products or other objects that they wish to identify.